tasks/README.md

```diff
@@ -0,0 +1,96 @@
+# Tasks
+
+## Task object
+
+```
+{
+  "type":"Dev | Group",
+  "links": [
+    <link1>,
+    <link2>
+  ],
+  "endsOn":"<end date>",
+  "startedOn":"<start date>",
+  "status": "Active",
+  "ownerId":"<app owner user id>",
+  "percentCompleted":10%,
+  "dependsOn": [
+    <tasksList>
+  ],
+  "participants": [list of users participating in this tasks. For 'group' type only],
+  "completionAward": { gold: 3, bronze: 300 },
+  "lossRate": { gold: 1 }  // Loss per day of overshoot after deadline
+}
+```
+
+## **Requests**
+
+|               Route                |    Description    |
+| :--------------------------------: | :---------------: |
+|      [GET /tasks](#get-tasks)      | Returns all tasks |
+|     [POST /tasks](#post-tasks)     | Creates new task  |
+| [PATCH /tasks/:id](#patch-tasksid) |   Updates tasks   |
+
+## **GET /tasks**
+
+> returns all the tasks
+
+- **Success Response:**
+- **Code:** 200
+  - **Content:**
+
+```
+{
+  message: 'Tasks returned successfully!'
+  tasks: [
+           {<task_object>},
+           {<task_object>}
+         ]
+}
+```
+
+- **Error Response:**
+  - **Code:** 503
+    - **Content:** `{ 'statusCode': 503, 'error': 'Service Unavailable', 'message': 'Something went wrong please contact admin' }`
+
+## **POST /tasks**
+
+- **Headers**  
+  Content-Type: application/json
+- **Body** `{ <task_object> }`
+- **Success Response:**
+- **Code:** 200
+  - **Content:**
+
+```
+{
+  message: 'Task created successfully!'
+  tasks: [
+           {<task_object>},
+           {<task_object>}
+         ]
```

Comment:
Only one task will be created per POST request. So task key will only contain one `task_object`.

Reply:
Please follow this pattern for the response, as this what we are using with all other Endpoints.
`{ "statusCode": 409, "error": "Conflict", "message": "User already exists" }`

Reply:
On the post request, do we only need to return the task that was created ?

Reply:
Yes, @sumitd94.
One request should always serve a single purpose.
For new updated data user will hit a separate GET request.